### ISSUE TITLE
cannot move out of borrowed context in loops inside match arms

### ISSUE THREAD

**Issue by drahnr** [djc/askama#222]
Template loops do not work when having a loop inside a match statement over the matched item
```rust
enum X {
    A(Vec<u8>),
    B
}
```

```
{% match x %}
{% when A with vektor %}
{% for item in vektor %}
{{ item }},
{% endfor %}
{% else %}
-
{% endmatch %}
```

I'll link a minimal reproduceable example in a bit.

**Comment by gregelenbaas** [comment]
I think this would be fixed by #221

**Comment by djc** [comment]
I'll close this as a duplicate of #107.
